### ISSUE TITLE
Feature request: variable to know folder of an existing mail

### ISSUE THREAD

**Issue by macdeport** [RealRaven2000/SmartTemplate4#103]
By example if such variable exists its value can be associated in many cases to a specific identity.

By the way in SmartTemplate⁴ what conditional can be used for such purpose?

**Comment by RealRaven2000** [comment]
I am only starting to get into "conditional" expressions at the moment - for 3.2 I added the first conditional function 
`%conditionalText(forwardMode,"text1","text2")%`
which adds 2 different test depending on whether the mail was forwarded inline or as attachment. The variables of SmartTemplates are not a complete programming language (almost each variable is executed independently of the next / previous one) - so I would need to see some of your use cases to come up with something that is useful.

**Comment by macdeport** [comment]
> I am only starting to get into "conditional" expressions at the moment - for 3.2 I added the first conditional function
%conditionalText(forwardMode,"text1","text2")%
which adds 2 different test depending on whether the mail was forwarded inline or as attachment.

Understanding (and appreciate) the philosophy but not clear about syntax if I have to use. Perhaps you mean something like
%conditionalText(forwardMode,"textInline","textAttachment")%

**Comment by RealRaven2000** [comment]
> > I am only starting to get into "conditional" expressions at the moment - for 3.2 I added the first conditional function
> > `%conditionalText(forwardMode,"text1","text2")%`
> > which adds 2 different test depending on whether the mail was forwarded inline or as attachment.
> 
> I notice this (#66 & #97) and appreciate conditionals to give to SmartTemplate⁴ an natural amplifier to its power.
> 
> Understanding (and appreciate) the philosophy but not clear about syntax if I have to use. Perhaps you mean something like
> `%conditionalText(forwardMode,"textInline","textAttachment")%`

Yes. And this one is purely for inserting texts based on conditions - which I can easily expand with other 1st parameters later without having too complicated syntax for the end users.

> Or a more generic and clear syntax (Python inspirited ;-) ) for this case of **2-fold incase** as :
> `%incase(forwardMode,{inline:"text1",attachment:"text2"})%`
I am not going to create a full programming language - the parser of SmartTemplates is very limited and based on regular expressions. I rather keep is simple for my users and for the parser's sake. 

Also what you want to do is switch identities (which is more akin to setting headers, like setting "to", "from" etc.) and not insert text, so this is in a different class anyway.

> By example, as in many cases a folder is strongly correlated to a specific identity, if such a function exists its value can be statistically associated to a given identity:
> `%incase(%mailFolder%,{folder1:"identity1",folder2:"identity2",folder3:"identity3"})%`

I think you almost have something like this in my Add-on[ QuickFolders](https://addons.thunderbird.net/en-US/thunderbird/addon/quickfolders-tabbed-folders/) where folders are representable as tabs in their own toolbar - this is a "Pro" function (supported once you get a QF Pro license)

![image](https://user-images.githubusercontent.com/6314674/103230697-feac6900-492d-11eb-9518-3ba11cd57788.png)
This basically changes the identity for a folder (and on demand also its sub folders). Disadvantage - you have to drag the folder to create a tab to configure that; but if needed you can hide the tab later via "Categories" so it's not going to take up all your screen space.

The function is located in the QuickFolders Commands on any QuickFolders tab under "_tab specific properties..._":
![image](https://user-images.githubusercontent.com/6314674/103230747-1b48a100-492e-11eb-98ee-af51704b1c46.png)

If you would like to test drive it I can send you a temporary license key, just drop me an email.

**Comment by macdeport** [comment]
This `[Tab-specific Properties]` functionality you kindly pointed out in richness of [QuickFolders](https://addons.thunderbird.net/en-US/thunderbird/addon/quickfolders-tabbed-folders/) Pro is exactly that I look for since a long time and unfortunately have not noticed: thanks for your Xmas gift :-)

> If you would like to test drive it I can send you a temporary license key, just drop me an email.

Thanks for your offer but I have already chosen to have the Pro licenses of QuickFolders and SmartTemplate⁴ as I appreciate the developer choices and results ;-)

**Comment by macdeport** [comment]
A SmartTemplate⁴ conditional can inscribe in reality the link between each folder and a potential specific template (kind of generalization/complementation of _QuickFolders_ "Tab-specific Properties"):

`%incase(%mailFolder%,{folder1:"template1",folder2:"template2",folder3:"template3"})%`

**Comment by RealRaven2000** [comment]
> A SmartTemplate⁴ conditional can inscribe in reality the link between each folder and a potential specific template (kind of generalization/complementation of _QuickFolders_ "Tab-specific Properties"):
> 
> `%incase(%mailFolder%,{folder1:"template1",folder2:"template2",folder3:"template3"})%`

I don't think using the variables in one template in order to configure locations of other templates is the correct approach. If we want folder specific templates, they should be configured at the folder (ideally in the folder context menu or as additional setting on the folder properties tab, e.g. like this. Generally I would prefer a UI solution and not doing a complicated variable that people will have difficulties using and understanding (even though it would mean more work for me).

First (bad) mockup:
![image](https://user-images.githubusercontent.com/6314674/103886180-08299400-50d9-11eb-91bc-253c05edaee3.png)

potentially not enough space there, and "General Information" is probably not the place / heading to put this under. Also we need 3 different templates (write / reply / forward) so we have the same UI problem that the SmartTemplates settings dialog already dealt with.

A dedicated non-tabbed folder dialog (configurable via right-click) might be a better choice:
![image](https://user-images.githubusercontent.com/6314674/103887662-5c357800-50db-11eb-90e7-1bb8fa23761d.png)

that would be easier to understand and use for any user, and would only be shown in the context of a specific folder.

**Comment by macdeport** [comment]
I agree with you avoiding multicase conditional for better et more ergonomic (comfortable) UI, but the main point is to organize a **strong synergy between _QF_ and _SM⁴_ if both are installed** without a too heavy interfacing for you : one way is perhaps to use both configuration JSON files...

**Comment by macdeport** [comment]
> This basically changes the identity for a folder (and on demand also its sub folders). Disadvantage - you have to drag the folder to create a tab to configure that; but if needed you can hide the tab later via "Categories" so it's not going to take up all your screen space.
I am learning to use categories: could you elaborate a little bit (or suggest the helping URL). Thank you.

**Comment by RealRaven2000** [comment]
> I am learning to use categories: could you elaborate a little bit (or suggest the helping URL). Thank you.

For a description on categories, check here - and look for the heading "More Tabs, more Functions"
https://quickfolders.org/index.html

**Comment by macdeport** [comment]
Suggestion: why not to put `#more_tabs_more_functions` in your HTML?

**Comment by RealRaven2000** [comment]
> Suggestion: why not to put `#more_tabs_more_functions` in your HTML?

I added some in-page anchors. please remember this is the SmartTemplates issue tracker, so don't piggyback too many QuickFolders suggestions here.
